src/github.com/matrix-org/dendrite/clientapi/routing/routing.go

```diff
@@ -0,0 +1,51 @@
+package routing
+
+import (
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/matrix-org/dendrite/clientapi/readers"
+	"github.com/matrix-org/dendrite/clientapi/writers"
+	"github.com/matrix-org/util"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const pathPrefixR0 = "/_matrix/client/r0"
+
+// Setup registers HTTP handlers with the given ServeMux. It also supplies the given http.Client
+// to clients which need to make outbound HTTP requests.
+func Setup(servMux *http.ServeMux, httpClient *http.Client) {
+	apiMux := mux.NewRouter()
+	r0mux := apiMux.PathPrefix(pathPrefixR0).Subrouter()
+	r0mux.Handle("/sync", make("sync", wrap(func(req *http.Request) (interface{}, *util.HTTPError) {
+		return readers.Sync(req)
+	})))
+	r0mux.Handle("/rooms/{roomID}/send/{eventType}",
+		make("send_message", wrap(func(req *http.Request) (interface{}, *util.HTTPError) {
+			vars := mux.Vars(req)
+			roomID := vars["roomID"]
```

Comment:
Can we inline the vars["<key>"] here? It doesn't look like we need to store these as local variables here.